trustmanager/keystore.go

```diff
@@ -40,7 +42,47 @@ func NewKeyFileStore(baseDir string, p notary.PassRetriever) (*GenericKeyStore,
 	if err != nil {
 		return nil, err
 	}
-	return NewGenericKeyStore(fileStore, p), nil
+	migrateTo0Dot4(fileStore)
+	store := NewGenericKeyStore(fileStore, p)
+	return store, nil
+}
+
+func migrateTo0Dot4(s Storage) {
+	for _, file := range s.ListFiles() {
```

Comment:
Instead of listing all files and checking on each one, can we instead check if the `notary.RootKeysSubdir` and `notary.NonRootKeysSubdir` exist and only migrate if keys live there?  And then we can delete those subdirectories once we've finished a successful migration